MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,899 @@
+<td>
+Generated for the Enum type property.
+</td>
+</tr>
+</table>
+
+You can customize the property settings or cancel the generation of `DataFormItem` by handling the `GenerateDataFormItem` event.
+
+## Cancel to generate data editor
+
+You can cancel the specific data editor by handling the `GenerateDataFormItem` event or by defining display attribute to avoid the particular data field being displayed.
+
+#### Using attributes
+
+You can set `AutoGenerateField` to `false` for canceling the data editor generation.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(AutoGenerateField = false)]
+    public int ID { get; set; } 
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+In the following code, the data editor generation for the `MiddleName` property is canceled by setting the `Cancel` property to true of the `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+            e.Cancel = true;
+
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Change the auto-generated editor settings
+
+You can change the data editor settings by using the property of `DataFormItem` in the `GenerateDataFormItem` event when AutoGenerate is true.
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "MiddleName")
+            {
+                e.DataFormItem.IsReadOnly = true;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+### Change the autogenerated editor visibility
+
+You can change the specific editor visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
```

Comment:
GenerateDataFormItem event when AutoGenerate is true.

For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor

remove

Reply:
removed